app/department.py

```diff
@@ -0,0 +1,108 @@
+import json
+import xml.etree.ElementTree as ET  # noqa: N817
+from abc import ABC, abstractmethod, abstractproperty
+from typing import Any
+
+
+class MainDepart(ABC):
+    def __init__(self, method: str = None, product: str = None) -> None:
+        self._method = method
+        self._product = product
+
+    @abstractmethod
+    def no_such_method(self) -> None:
```

Comment:
In the base class MainDepart, the no_such_method method is implemented but then overridden in every subclass. Since the method in the base class is abstract, there’s no need to provide an implementation there. This implementation can be removed from MainDepart.